androidsdk/src/main/java/com/wootric/androidsdk/views/phone/SurveyLayoutPhone.java

```diff
@@ -317,6 +323,44 @@ private void setTexts() {
         mEtFeedback.setHint(mSettings.getFollowupPlaceholder(mRatingBar.getSelectedScore()));
     }
 
+    private void setColors() {
+        Resources res = mContext.getResources();
+        mColorSelected = res.getColor(mSettings.getScoreColor());
+        mColorEnabled = res.getColor(mSettings.getSurveyColor());
+        mRatingBar.setSelectedColor(mColorSelected);
+
+        mBtnEditScore.setTextColor(mColorEnabled);
+        mBtnDismiss.setTextColor(mColorEnabled);
+
+        mTvSurveyHeader.setBackgroundColor(mColorEnabled);
+
+        setCursorDrawableColor(mEtFeedback, mColorSelected);
+    }
+
+    private static void setCursorDrawableColor(EditText editText, int color) {
+        try {
+            Field fCursorDrawableRes = TextView.class.getDeclaredField("mCursorDrawableRes");
+            fCursorDrawableRes.setAccessible(true);
+            int mCursorDrawableRes = fCursorDrawableRes.getInt(editText);
+            Field fEditor = TextView.class.getDeclaredField("mEditor");
+            fEditor.setAccessible(true);
+            Object editor = fEditor.get(editText);
+            Class<?> clazz = editor.getClass();
```

Comment:
It's actually kind of a hacky solution to change the EditText's cursor. It's done by using Reflection with Java:
https://docs.oracle.com/javase/tutorial/reflect/
(I just learned about Reflection when searching StackOverflow for a solution on how to change the Cursor's color, so it's a new concept for me :smile: )

Android API doesn't have a public method to change the cursor's color at runtime. So basically this whole method goes through the TextView class looking for the Cursor drawable so it can change its color at runtime using reflection.

I think C# has a similar concept too, but I don't really know C# that much.


Class<?> is used when the class to be modeled is unknown. Now, I do know which class `editor.getClass()` will return, is `Editor.java`, but that class is hidden, so I can't import that class.